frontend/public/components/utils/_number-spinner.scss

```diff
@@ -23,3 +23,6 @@
   top: 4px;
 }
 
+.pf-c-input-group > .co-m-number-spinner {
+  margin-right: 20px;
```

Comment:
Since you really only need the spacing when an item follows a number spinner, I recommend adding margin-left to the unit dropdown.

Additionally, as a general rule, we try to avoid descendent (in this case, child) selectors like this one since they are reliant on the markup.